util/execdetails/execdetails.go

```diff
@@ -253,11 +253,47 @@ func (crs *CopRuntimeStats) String() string {
 		procTimes[n-1], procTimes[0], procTimes[n*4/5], procTimes[n*19/20], totalRows, totalIters, totalTasks)
 }
 
+// ReaderRuntimeStats collects stats for TableReader, IndexReader and IndexLookupReader
+type ReaderRuntimeStats struct {
+	sync.Mutex
+
+	copRespTime []time.Duration
```

Comment:
How about to:
1. put the response time into `CopRuntimeStats`
2. rename `CopRuntimeStats` to `CopStatsColl`, because now it contains the response time, not only the runtime stats.

For example, we can:
```go
type CopStats struct {
    // RuntimeStats records the runtime statistics of a cop task executed in
    // the tikv coprocessor.
    //
    // NOTE: The time it's recorded is the cop task proc time.
    RuntimeStats

    // respTime records the response time of a cop task.
    //
    // NOTE: This time is composed of:
    // 1. TiKV side: wait time, proc time, encode time
    // 2. network time
    // 3. task retry, if any
    // 3. TiDB side: decode time
    respTime time.Duration
}
```

```go
// CopStatsColl collects all the CopStats belonging to the same tikv store.
type CopStatsColl struct {
	sync.Mutex

	// stats stores the runtime statistics of coprocessor tasks.
	// The key of the map is the tikv-server address. Because a tikv-server can
	// have many region leaders, several coprocessor tasks can be sent to the
	// same tikv-server instance. We have to use a list to maintain all tasks
	// executed on each instance.
	stats map[string][]*CopStats
}
```

Reply:
I prefer not to put it into the `CopRuntimeStats`. Because the `CopRuntimeStats` only shows in cop task and `ReaderRuntimeStats` only shows in reader root task. It's easy for each plan to scale.